alexandria/core/api.py

```diff
@@ -0,0 +1,66 @@
+from django.conf import settings
+from django.core.exceptions import ValidationError
+
+from alexandria.core import models
+
+
+def create_document_file(user, group, document_attributes, file_attributes):
+    """
+    Create a document and file with the given attributes.
+
+    This function eases the creation of documents and files by automatically setting important fields.
+    Uses ``create_file`` to create the file.
+
+    :param user: The user creating the document and file.
+    :param group: The group creating the document and file.
+    :param document_attributes: A dictionary containing the fields for the document. (required: category, title)
+    :param file_attributes: A dictionary containing the fields for the file. (required: name, content, mime_type, size)
+    :return: A tuple containing the created document and file.
```

Comment:
I think the reST syntax is obsolete. Instead, use type annotations where possible, and just plain text description of the arguments. For example here, it's not clear whether the user and group params are strings or objects.